### Given
Density of bacteria in plate = $\rho= 10/cm^2$
Area $A=10cm^2$
Probability of a bacteria carrying mutation = $\mu$


### Facts

Bacteria on plate are like 'white and black' balls in a box with thw white ones representing the mutated copies.
Thus, assuming total number $N \sim Poisson(\lambda)$ and the coordinates $(X_i,Y_i)$ to be in a uniform square(assume the 'dish' to be square ) it is safe to assume, that this is a poisson point process with intensity $\mu$ and the mutated 'whites' make it a poisson coloring problem.

Hence, if $MW$ represents the mutated whites and $UB$ represent the unmutated black balls, we have 

$MW \sim PPP(\mu\rho)$ and $UB \sim PPP((1-\mu)\rho)$


Thus for part (a): P(probability that n 'mutated whites' exist) = $\frac{e^{-\rho\mu}(\mu\rho)^n}{n!}$



### Part (b)

In [12]:
from math import log, exp, e
t_min_grow = log(10)/log(1.05)
rho = 10
A = 10
N = rho*A


In [13]:
print t_min_grow

47.1936328191


Radius of cell growing for last $t$ minutes: $1.05^t \times 10^{-3}cm$. Thus for a cell to form a detectable cluster it should grow for atleast: $\frac{log(10)}{log(1.05}=47.19$ minutes

And the total time for grow is $60$ minutes, so that MAX wait time before growth starts is $60-47.19=12.81$ minutes

In [19]:
delta = (1-exp(-12.81/20))
print(delta)

0.472971156268


$\delta = p(\text{wait time $\leq$ 12.81 minutes}) = 1-\exp(-12.81/20) = 0.473$


So now we need these criteria for calling a ball white: 
- It gets mutated (PPP)
- It has a waiting time of <12.81 minutes

So we have a new $PPP(\rho \mu \delta)$ and 
$\frac{17}{\rho \times A} = \exp(-\rho \mu \delta) \times \frac{(\rho \mu \delta)^{17}}{17!}$

$\ln(\frac{17}{\rho A}) = -\rho \mu \delta + 17 \ln(\rho \mu \delta) - \ln(17!)$

In [25]:
from sympy import solve, Eq, symbols
from mpmath import log as mpl
from math import factorial
mu= symbols('mu')
lhs = log(17/100.0,e)
rhs = Eq(-rho*mu*delta + 17*(rho*mu*delta-1) -mpl(factorial(17)))
s = solve(rhs, mu)


In [26]:
print (s)

[0.667391034061759]


Thus, approximated $\mu = 0.667$ for the given dataset!